addon/async-test-waiter.ts

```diff
@@ -13,12 +13,13 @@ function getNextToken(): Token {
  * @public
  * @class TestWaiter<T>
  */
-export default class TestWaiter<T extends object | Primitive = Token> implements ITestWaiter<T> {
+export default class AsyncTestWaiter<T extends object | Primitive = Token>
```

Comment:
Let's call this `TestWaiterImpl` (IMHO adding "async" doesn't _really_ add any value), but rename the imported interface locally:

Reply:
Uggg. That feels so Java-y. It's offending my sensibilities. I'll do it thought since we're planning on essentially making it private-ish. I don't care as much about the name here, as it's not public.